typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -44,41 +67,65 @@ export class EvmERC20WarpRouteReader {
    */
   async deriveWarpRouteConfig(
     address: Address,
-  ): Promise<DerivedERC20WarpRouteConfig> {
+  ): Promise<DerivedTokenRouterConfig> {
+    // Derive the config type
+    const type = await this.deriveTokenType(address);
     const fetchedBaseMetadata = await this.fetchBaseMetadata(address);
-    const fetchedTokenMetadata = await this.fetchTokenMetadata(
-      fetchedBaseMetadata.token,
-    );
+    const fetchedTokenMetadata = await this.fetchTokenMetadata(type, address);
 
-    const results: DerivedERC20WarpRouteConfig = {
+    return {
       ...fetchedBaseMetadata,
       ...fetchedTokenMetadata,
+      type,
+    } as DerivedTokenRouterConfig;
+  }
+
+  /**
+   * Derives the token type for a given Warp Route address using specific methods
+   *
+   * @param address - The Warp Route address to derive the token type for.
+   * @returns The derived token type, which can be one of: collateralVault, collateral, native, or synthetic.
+   */
+  async deriveTokenType(address: Address): Promise<DerivedTokenType> {
```

Comment:
nit: maybe the argument is called `warpRouteAddress` so that i'm not confused about whether it's the token address or the warp route

Reply:
Fixed.